src/com/rcv/ResultsWriter.java

```diff
@@ -116,12 +133,17 @@ void generatePrecinctSummarySpreadsheets(
   ) {
     Set<String> filenames = new HashSet<>();
     for (String precinct : precinctRoundTallies.keySet()) {
-      // filename is the path to write the output file
-      String filename = getPrecinctFilename(precinct, filenames) + ".xlsx";
+      // precinctFileString is a unique filesystem-safe string which can be used for creating
+      // the precinct output filename
+      String precinctFileString = getPrecinctFileString(precinct, filenames);
+      // filename for output
+      String outputFileName = String.format("%s_%s.xlsx", this.timestamp, precinctFileString);
```

Comment:
Apply outcome of above comment here, e.g. adding "visualizer_spreadsheet" into the mix.